src/components/Pressable/PressableWithFeedback.js

```diff
@@ -24,17 +25,41 @@ const PressableWithFeedbackDefaultProps = {
 
 const PressableWithFeedback = forwardRef((props, ref) => {
     const propsWithoutStyling = _.omit(props, omittedProps);
+    const [disabled, setDisabled] = useState(props.disabled);
+
+    useEffect(() => {
+        setDisabled(props.disabled);
+    }, [props.disabled]);
+
     return (
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        <GenericPressable ref={ref} style={props.wrapperStyle} {...propsWithoutStyling}>
+        <GenericPressable
+            ref={ref}
+            style={props.wrapperStyle}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...propsWithoutStyling}
+            disabled={disabled}
+            onPress={(e) => {
+                setDisabled(true);
+                const onPress = props.onPress(e);
+                InteractionManager.runAfterInteractions(() => {
+                    if (!(onPress instanceof Promise)) {
```

Comment:
nab, but if onPress is not an instance of Promise, does it still have to be inside `InteractionManager`?

could you elaborate difference between the above code and the following:

```
const onPress = props.onPress(e);
if (!(onPress instanceof Promise)) {
   setDisabled(props.disabled);
   return;
}

InteractionManager.runAfterInteractions(() =>
  onPress.then(() => {
                          setDisabled(props.disabled);
                      });
});
```

Reply:
> on some android devices, there is still a possibility for it to capture multiple interactions before onPress's completion

I am not sure about this when we have sync calls, but it's okay to keep what it is right now.